tests/StateSnapshot.test.js

```diff
@@ -298,6 +298,20 @@ ruleTester.run('state-snapshot-rule', rule, {
       </>
     );
   })`,
+    `
+  function useExample2(s) {
+    const {b: {c} } = useSnapshot(s.a1);
+    const val = useMemo(()=>{
+      return s.a1.b.c
+    },[s.a1.b.c])
+  }`,
```

Comment:
That's actually not good.
The deps should come from snapshots.
```js
  function useExample2(s) {
    const {b: {c} } = useSnapshot(s.a1);
    const val = useMemo(()=>{
      return c
    },[c])
  }
```

Reply:
Updated